operators/advanced_network/Dockerfile

```diff
@@ -142,25 +122,107 @@ RUN python3 -m pip install --no-cache-dir \
+    && DISTRO=$(. /etc/os-release && echo ${ID}${VERSION_ID}) \
+    && DOCA_REPO_LINK=https://linux.mellanox.com/public/repo/doca/${DOCA_VERSION}/${DISTRO}/${DOCA_ARCH} \
+    && echo "Using DOCA_REPO_LINK=${DOCA_REPO_LINK}" \
+    && LOCAL_GPG_KEY_PATH="/usr/share/keyrings/mellanox-archive-keyring.gpg" \
+    && curl -fsSL ${DOCA_REPO_LINK}/GPG-KEY-Mellanox.pub | gpg --dearmor | tee ${LOCAL_GPG_KEY_PATH} > /dev/null \
+    && echo "deb [signed-by=${LOCAL_GPG_KEY_PATH}] ${DOCA_REPO_LINK} ./" | tee /etc/apt/sources.list.d/mellanox.list
+
+# Install mlnx-ofed-kernel-utils (utilities including ibdev2netdev used by tune_system.py)
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        mlnx-ofed-kernel-utils \
+    && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /opt
+
 # ==============================
-# DOCA Target
-# This stage is only built when --target doca is specified. It contains any DOCA-specific configurations.
+# DPDK Target (Default)
+# This is the default target for users who only need DPDK support.
+# GPUNetIO/DOCA is NOT included.
 # ==============================
-FROM common-deps AS doca
-# DOCA-specific installation or steps (if needed)
+FROM dpdk-build AS dpdk
+# DPDK is ready to use, no additional steps needed
+
+# ==============================================================
+# gpunetio-deps: Add DOCA SDK packages for GPUNetIO support
+# ==============================================================
+FROM dpdk-build AS gpunetio-deps
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+# Install DOCA SDK packages required for GPUNetIO
+# (DOCA repo is already configured in dpdk-build stage)
+# - libdoca-sdk-dma-dev: for gpunetio backend (dependency of doca-gpunetio module)
+# - libdoca-sdk-gpunetio-dev: for gpunetio backend (doca-gpunetio module)
+# - libdoca-sdk-eth-dev: for gpunetio backend (doca-eth module)
+# - libdoca-sdk-flow-dev: for gpunetio backend (doca-flow module)
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        libdoca-sdk-dma-dev \
+        libdoca-sdk-gpunetio-dev \
+        libdoca-sdk-eth-dev \
+        libdoca-sdk-flow-dev \
+    && rm -rf /var/lib/apt/lists/*
 
 # ==============================
-# DPDK Target
-# This stage is only built when --target dpdk is specified. It contains any DPDK-specific configurations.
+# GPUNetIO Target
+# This stage is built when --target gpunetio is specified.
+# It includes DOCA packages required for GPUNetIO support.
 # ==============================
-FROM common-deps AS dpdk
-# DPDK-specific installation or steps (if needed)
+FROM gpunetio-deps AS gpunetio
+# GPUNetIO is ready to use with DOCA support
 
 # ==============================
 # Rivermax Target
 # This stage is only built when --target rivermax is specified. It installs and configures Rivermax SDK.
+# Note: Rivermax does not require DPDK - extends from base-deps directly.
 # ==============================
-FROM common-deps AS rivermax
+FROM base-deps AS rivermax
+
+ARG TARGETARCH
```

Comment:
should come from base ?

```suggestion
```

Reply:
It's a docker build option so I don't think it can be inherited?